### ISSUE TITLE
IRC connection using ipv6

### ISSUE THREAD

**Issue by Miniontoby** [reactiflux/discord-irc#597]
Hey,

I am trying to get my bot connected using ipv6 so I can get my vhost, but it doesnt seems to work.

Is there a way how to get this working or is this not supported?

- Miniontoby

**Comment by aronson** [comment]
Hello, the library used by this discord-irc project does not support IPv4.

I rewrote a large chunk of the project and upgraded to a new IRC library that supports IPv6.
I just tested IPv6 for you. It works on my fork.

My fork is here: https://github.com/aronson/discord-irc
The config format changed slightly. The app will give you errors to guide you through the changes or you can review the README.

The fork is used in production in a Discord server of 15k users and has seen some testing.

**Comment by Miniontoby** [comment]
How to run that?
I cannot use deno on my OS...

**Comment by aronson** [comment]
You can run the compiled program -- it has no dependencies. Don't need deno or docker.

Get the right binary for your system from [the latest release](https://github.com/aronson/discord-irc/releases/)

The [README](https://github.com/aronson/discord-irc#native-builds-easiest) has a section on how to run it.

Pretty much just
- download program
- put config.json next to it
- run it

I only know one server using my fork right now. If you have questions or run into issues feel free to reach out

**Comment by Miniontoby** [comment]
Those binaries don't work for my OS. I've got OpenBSD and the linux binaries don't work (they have completely messed up my shell as well [now it shows like random utf-8 chars] so I had to start a new terminal session...)

**Comment by aronson** [comment]
Ah, sorry to mess up your terminal! I see there's OpenBSD support for Deno I can download. I'll get this compiled for you and send you an update when I've got a working binary. We might be able to add OpenBSD to our auto build list in general.

Thanks for making me aware of this OS needing support 😄

**Comment by Miniontoby** [comment]
Yeah, but tbh it would be nice if it would just be possible to run it with just standard nodejs and/or tsc

**Comment by aronson** [comment]
Deno binaries for the framework itself can be found on OpenBSD through [this link](https://openports.pl/path/lang/deno) as far as I'm aware if you don't want to wait on me.

If you can download my binaries, you can download this, no? 😉 Deno is a statically compiled, standalone executable on many platforms.

Sorry, no node support. I use Deno-specific libraries, such as the IRC one with IPv6, and a large section of the codebase would need to be re-written, yet again, not to mention dealing with issues using Deno libraries under node.

The creator of node abandoned it to create and work on Deno. This is the key driver of my decision.

Of course if a kind Samaritan solves the import mapping, crafts another package.json, figures out Deno->Node backwards compat, rewrites deps.ts (fixing all import paths), re-configures Deno to allow this, and can provide me with a PR, it'll get merged.

**Comment by Miniontoby** [comment]
That port is not in the `pkg` list yet and when I try building it myself it gives me an error and idk what to do with it...

Meaning that I cannot install/run deno myself. So yeah, I guess I will wait and/or continue using this discord-irc repo

**Comment by aronson** [comment]
I investigated generating a binary and found I cannot support OpenBSD builds with an automatic runner at this time.

Deno is available for OpenBSD. 64 bit Linux binaries such as I provide are also runnable on FreeBSD.

If you are having trouble, I suggest returning to OpenBSD resources on how to install custom software.
There are many people running Deno themselves on OpenBSD.

Perplexity.AI generated [a guide on how to do so](https://www.perplexity.ai/search/How-do-I-YJ5GjbvAQJaj4q.KLr2egw?s=c ).
It is missing information. If you want to run this software, you’ll need to go hunting.

If you would like IPv6 support, it’s out there.

**Comment by Miniontoby** [comment]
I know how to install software! But the ports link you send is NOT in the pkg list yet! So I cannot install it!

I will still try my best...

**Comment by aronson** [comment]
```bash
git clone https://git.FreeBSD.org/ports.git /usr/ports
pkg install -y www/deno
```

Works for me on FreeBSD. [The OpenBSD process is similar but different](https://www.openbsd.org/faq/ports/ports.html#PortsFetch).

**Comment by aronson** [comment]
[We have a new release out that includes a FreeBSD binary.](https://github.com/aronson/discord-irc/releases/)

**Comment by aronson** [comment]
On OpenBSD 7.4 all I had to do was:
```bash
# As root
pkg_add deno
pkg_add git
# As regular user
git clone https://github.com/aronson/discord-irc.git
cp config.json discord-irc/config.json
cd discord-irc/
deno task start
```

Are you running an older version?

For some reason `deno task compile` on the machine isn't emitting a usable binary. It works fine run from source.

<img width="840" alt="image" src="https://github.com/reactiflux/discord-irc/assets/1026348/8fdd5198-61ff-47f8-bb27-379b5e819d68">

**Comment by Miniontoby** [comment]
> [We have a new release out that includes a FreeBSD binary.](https://github.com/aronson/discord-irc/releases/)

You cannot use FreeBSD binaries on OpenBSD.


> On OpenBSD 7.4 all I had to do was:

I am still on OpenBSD 7.2... I will have to schedule an OS upgrade, I guess...

**Comment by aronson** [comment]
7.2 reaches end of life in 2 weeks and 2 days. I'm not surprised it didn't see the deno update.

Just checked, and 7.3 doesn't have it either. Seems to be included in 7.4 `-current` specifically.
